categorize_notify/program.py

```diff
@@ -10,60 +10,86 @@
 
 import base64
 import time
+from datetime import datetime, timezone
 
 import autokitteh
 from autokitteh import google, openai, slack
 
 
 SLACK_CHANNELS = ["demos", "engineering", "ui"]
 
+processed_message_ids = set()
```

Comment:
why are these global?

Reply:
I tend to prefer the global approach for simplicity - the local approach requires passing it, understanding Python mutability of passed sets, etc.